### PR TITLE
Week 2 chris frey

### PR DESCRIPTION
**Pull request by Chris-Frey**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Coding Challenges
- [ ] Coding Concepts
- [ ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:
I may be misunderstanding hoisting, or perhaps didn't go deep enough. If that's the case, let me know and I'll go read more on it. Or any feedback generally is welcome.  Thanks for looking!

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -27,6 +27,17 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("divisible", () => {
+  it("returns a string stating whether or not an input is divisible by 3", () => {
+
+    expect(divisible(object1.number)).toEqual(`${object1.number} is divisible by three`)
```
Comment (chrisyhlee) [inline code-challenges.test.js:33]: Remember that the prompt tells you that your function is taking an object as an argument, not a value in that object. Also, in your toEqual, it is best to put exactly what the output of each variable should be, for precise testing.

File: code-challenges.test.js [hunk 2]
```diff
@@ -36,25 +47,52 @@ const object3 = { number: -7 }
...
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("capWords", () => {
+  it("returns all words capitalized", () => {
+
+    expect(capWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    
+    expect(capWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
```
Comment (chrisyhlee) [inline code-challenges.test.js:68]: Great test!

File: code-challenges.test.js [hunk 3]
```diff
@@ -36,25 +47,52 @@ const object3 = { number: -7 }
 
 // b) Create the function that makes the test pass.
 
+const divisible = (input) => {
```
Comment (chrisyhlee) [inline code-challenges.test.js:50]: Refer to above comment about what the argument of the function should be.

File: code-challenges.test.js [hunk 4]
```diff
@@ -36,25 +47,52 @@ const object3 = { number: -7 }
...
 
 // b) Create the function that makes the test pass.
-
+const capWords = (input) => {
+  return newArr = input.map(word => word[0].toUpperCase() + word.slice(1))
+}
 // Pseudo code:
+// create function with 1 input 
+// iterate through function with map 
+// call toUpperCase on first index of each word
```
Comment (chrisyhlee) [inline code-challenges.test.js:83]: No mention of .slice in your pseudo code

File: code-challenges.test.js [hunk 5]
```diff
@@ -63,5 +101,16 @@ const vowelTester3 = "challenges"
...
+  for(i = 0; i < str.length; ++i) {
+    if(vowels.includes(str[i])) {
+      return i
+    }
+  };
+}
 // Pseudo code:
+// create function with 1 input 
+// create variable array that holds all vowels 
+// forEach loop that stops when includes method === true
```
Comment (chrisyhlee) [inline code-challenges.test.js:115]: You said forEach here when the code is using a for loop, make sure your code reflects your pseudo

File: interview-questions.md [hunk 6]
```diff
@@ -8,50 +8,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: Methods are considered properties of objects that contain a function definition.  Basically methods are functions that are stored as properties.
 
-6. STRETCH: What is hoisting in JavaScript?
+5. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer:I *believe* hoisting is the ability to call a function before it has been defined in the code.  e.g. invoking a function on line 5 of your code, but having written the function on line 30.  JavaScript conceptually 'hoists' it up to the top to run the code.
 
-Researched answer:
+Researched answer: Hoisting is JavaScript moving all declarations to the top of the current scope.  This makes it a good rule to declare all variables at the top of their respective scopes.  Hoisting with `var` initializes the value to `undefined`. Hoisting `let` and `const` are not initialized and will result in an error.
```
Comment (chrisyhlee) [inline interview-questions.md:46]: You are correct! Hoisting is generally bad practice, as it could lead to potential issues. This is part of the reason why in modern JavaScript, it is recommended to use 'let' and 'const' instead of 'var', and those two are block-scoped declarations while 'var' is a function-scoped declaration.

TLDR: declare your variables and functions before you use them haha

**Review by chrisyhlee — APPROVED** [review]
Nice work on assessment 2! I suggest you work on your pseudo code, as some of it does not match what is actually happening in your code. Make sure your pseudo code explains each step your code takes to get from input to output, and how it's doing what it is doing. Overall, this displays Advanced Beginner, keep up the good work!

File: code-challenges.test.js (hunk 1 above)
Comment (Chris-Frey) [reply code-challenges.test.js:33]: Understood. I should have worked with the value inside the function instead of trying to hard-code it all the way through. Thank you!

File: code-challenges.test.js (hunk 5 above)
Comment (Chris-Frey) [reply code-challenges.test.js:115]: gotchya. I wrote the pseudo then did the code and never went back. I'll be more careful next time.

File: interview-questions.md (hunk 6 above)
Comment (Chris-Frey) [reply interview-questions.md:46]: Thanks for this feedback :)
